packages/react-router-dom/modules/Link.js

```diff
@@ -52,9 +57,24 @@ class Link extends React.Component {
     }
   }
 
-  render() {
-    const { replace, to, ...props } = this.props // eslint-disable-line no-unused-vars
+  absolute() {
+    const { to } = this.props
+    const { match } = this.context.router
+    const base = (match && match.url) ? match.url : ''
+    return resolveLocation(to, base)
+  }
+
+  componentWillMount() {
+    this.unlisten = this.context.router.listen(() => { this.forceUpdate() })
```

Comment:
Yes but a HOC will cost more resources than 2 functions per component, and typing `listen` is way easier than using a Component-with-function-children 🙈